build-logic/src/main/kotlin/ktlint-publication.gradle.kts

```diff
@@ -15,14 +15,11 @@ project.version =
     } else {
         definedVersion
     }
-// TODO: Remove `localGradleProperty` once https://github.com/gradle/gradle/issues/23572 or https://github.com/pinterest/ktlint/issues/1931 is fixed.
-project.group = localGradleProperty("GROUP").orNull ?: providers.gradleProperty("GROUP").orNull
-    ?: throw GradleException("Project group property is missing")
```

Comment:
We can still use `project.group = providers.gradleProperty("GROUP").get()` here, to align with `project.version`.